framework/db/BatchQueryResult.php

```diff
@@ -160,8 +173,14 @@ protected function getRows()
         $count = 0;
 
         try {
-            while ($count++ < $this->batchSize && ($row = $this->_dataReader->read())) {
-                $rows[] = $row;
+            while ($count++ < $this->batchSize) {
+                if ($row = $this->_dataReader->read()) {
+                    $rows[] = $row;
+                } else {
+                    // we've reached the end
+                    $this->trigger(self::EVENT_FINISH);
```

Comment:
I'm not sure if this is worth to be repeated conditionally outside the loop in case the read data size matches the batchSize or will it be picked up on the next iteration anyway?

Reply:
It will get hit on the next iteration, and `$rows` will just be an empty array.

![A screenshot of PhpStorm with an Xdebug breakpoint stopped at the EVENT_FINISH line, showing that $rows is an empty array](https://user-images.githubusercontent.com/47792/103418971-d7c87f80-4b45-11eb-947b-004448f41aa5.png)